converters/toZigbee.js

```
@@ -2543,6 +2543,94 @@ const converters = {
+                break;
+            }
+        },
+    },
+    tuya_thermostat_weekly_schedule: {
+        key: ['weekly_schedule'],
+        convertSet: async (entity, key, value, meta) => {
+            const maxTransitions = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleMaxTransitions');
+            const supportedModes = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleSupportedModes');
+            const firstDayDpId = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleFirstDayDpId');
+
+            function transitionToData(transition) {
+                // Later it is possible to move converter to meta or to other place outside if other type of converter
+                // will be needed for other device. Currently this converter is based on ETOP HT-08 thermostat.
+                // see also fromZigbee.tuya_thermostat_weekly_schedule()
+                const minutesSinceMidnight = transition.transitionTime;
+                const heatSetpoint = Math.floor(transition.heatSetpoint * 10);
+                return [
+                    (minutesSinceMidnight & 0xff00) >> 8,
+                    minutesSinceMidnight & 0xff,
+                    (heatSetpoint & 0xff00) >> 8,
+                    heatSetpoint & 0xff,
+                ];
+            }
+            for (const [, daySchedule] of Object.entries(value)) {
+                const dayofweek = parseInt(daySchedule.dayofweek);
+                const numoftrans = parseInt(daySchedule.numoftrans);
+                let transitions = [...daySchedule.transitions];
+                const mode = parseInt(daySchedule.mode);
+                if (!supportedModes.includes(mode)) {
+                    throw new Error(`Invalid mode: ${mode} for device ${meta.options.friendlyName}`);
+                }
+                if (numoftrans != transitions.length) {
+                    meta.logger.warn(`Invalid numoftrans provided. Real: ${transitions.length} ` +
```

I think it would be better to throw an exception here instead of a warning